### DISCUSSION TITLE
gfortran-11  killed: 9

### DISCUSSION THREAD

**Discussion by mcuendet** [Homebrew/discussions#2386] [Everyday usage] [answered]
Hi,

I'm trying to install gcc on a MacBook Pro M1 running OSX 11.6.1. Everything runs smoothly, but gfortran gets killed right away ```Killed: 9```. This may have to do with binary signature. Note that gcc itself and other associated binaries work. 

I tried different things like removing the binary/restarting/re-installing, but I still get the same error. Any idea?

Thanks, 
Michel

Crash report: 

```Process:               gfortran-11 [43155]
Path:                  /Users/USER/*/gfortran-11
Identifier:            gfortran-11
Version:               ???
Code Type:             ARM-64 (Native)
Parent Process:        bash [793]
Responsible:           Terminal [789]
User ID:               501

Date/Time:             2021-11-01 09:45:23.844 +0100
OS Version:            macOS 11.6.1 (20G224)
Report Version:        12
Anonymous UUID:        47350536-C2EA-36A5-C324-A98CA3137BB5


Time Awake Since Boot: 1500 seconds

System Integrity Protection: enabled

Crashed Thread:        Unknown

Exception Type:        EXC_BAD_ACCESS (Code Signature Invalid)
Exception Codes:       0x0000000000000032, 0x00000001008a4000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x2

kernel messages:

VM Regions Near 0x1008a4000:
--> mapped file                 1008a4000-1009a0000    [ 1008K] r-x/r-x SM=COW  Object_id=61c20ad
    mapped file                 1009a0000-1009dc000    [  240K] rw-/rw- SM=COW  Object_id=61c20ad

Backtrace not available

Unknown thread crashed with ARM Thread State (64-bit):
    x0: 0x00000001008a4000   x1: 0x000000016f55b618   x2: 0x000000016f55b5c8   x3: 0x0000000100caa264
    x4: 0x0000000000000070   x5: 0x0000000000000073   x6: 0x000000016f55b370   x7: 0x0000000000000020
    x8: 0x000000016f55b5f8   x9: 0x000000016f55b5d8  x10: 0x0000000100ccea78  x11: 0x0000000000000009
   x12: 0x0000000000000009  x13: 0x0000000000000002  x14: 0x0000000000000000  x15: 0x0000000000000000
   x16: 0x0000000100c7e3f0  x17: 0x6ae100016f55b5c8  x18: 0x0000000000000000  x19: 0x000000016f55b618
   x20: 0x00000001008a4000  x21: 0x000000016f55b5c8  x22: 0x00000001008a4000  x23: 0x000000016f55b778
   x24: 0x000000016f55b860  x25: 0x000000016f55b6b0  x26: 0x0000000000000000  x27: 0x0000000000000000
   x28: 0x000000016f55b758   fp: 0x000000016f55b5b0   lr: 0x6f6b800100c7e3b8
    sp: 0x000000016f55b530   pc: 0x0000000100c807dc cpsr: 0x80000000
   far: 0x00000001008a4000  esr: 0x92000007


Binary images description not available


External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
Writable regions: Total=8400K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8400K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8176K        1 
VM_ALLOCATE                        1.0G        2 
VM_ALLOCATE (reserved)              32K        2         reserved VM address space (unallocated)
mapped file                       2576K        7 
===========                     =======  ======= 
TOTAL                              1.1G       13 
TOTAL, minus reserved VM space     1.1G       13 

```

**Comment by carlocab** [comment, marked as answer]
Try
```
brew reinstall $(brew deps gcc) gcc
```

**Reply by mcuendet** [reply]
Hi carlocab, 

Thanks for the quick suggestion! But it didn't help in my case... Still killed:9.

Any other idea?

**Reply by mcuendet** [reply]
By the way:
```
> spctl -a  -v /Users/*/src/homebrew/Cellar/gcc/11.2.0/bin/gfortran-11
[...]
/Users/*/src/homebrew/Cellar/gcc/11.2.0/bin/gfortran-11: invalid signature (code or signature have been modified)
```

**Reply by carlocab** [reply]
No idea, sorry. Do you get the same issue with an installation of Homebrew into `/opt/homebrew`?

**Reply by mcuendet** [reply]
Hi, 

This actually worked. I just re-installed everything in /opt/homebrew. That means there is a bug somewhere in the gcc installation scripts when a custom location is used...

Thanks again for your help, 
Michel

**Reply by carlocab** [reply]
Did you build GCC from source, or did you use `--force-bottle` to force use of the GCC bottle?

**Reply by mcuendet** [reply]
No I simply used ```brew install gcc```.  Thanks.

**Reply by IgnusG** [reply]
I have the same problem. Unfortunately I cannot use the global install since it's a shared Mac. 

I've used the same command as @mcuendet and gcc installs correctly but gfortran closes immediately, installing `openmpi` fails on configure since it tries compiling with fortran and then reports an error that the output is not executable.

**Reply by astroboylrx** [reply]
I also have the same problem. I prefer to put `homebrew` under my home directory so installing stuff does not need alleviated user privilege. It is also stated in the [HomeBrew site ](brew.sh) that: "Homebrew won’t install files outside its prefix and you can place a Homebrew installation wherever you like." If that's true, then it means something may be hardcoded to `/opt/homebrew` when `brew` install `gcc-11`. Any suggestions or new ideas regarding possible fix?

**Reply by SMillerDev** [reply]
> I prefer to put homebrew under my home directory so installing stuff does not need alleviated user privilege.

Homebrew never changes user privilege levels except on the initial install.

> Homebrew won’t install files outside its prefix and you can place a Homebrew installation wherever you like.

The install instructions also say this, which is a pretty important note about that method:

> However do yourself a favour and use the installer to install to the default prefix. Some things may not build when installed elsewhere. One of the reasons Homebrew just works relative to the competition is because we recommend installing here. Pick another prefix at your peril!

https://docs.brew.sh/Installation
